app/user_test.go

```diff
@@ -515,6 +515,44 @@ func TestUpdateUserEmail(t *testing.T) {
 		assert.Equal(t, err.Id, "store.sql_user.update.email_taken.app_error")
 		assert.Nil(t, user3)
 	})
+
+	t.Run("Only the last token works if verification is required", func(t *testing.T) {
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.EmailSettings.RequireEmailVerification = true
+		})
+
+		// we update the email a first time and update. The first
+		// token is sent with the email
+		user.Email = th.MakeEmail()
+		_, appErr := th.App.UpdateUser(user, true)
+		require.Nil(t, appErr)
+
+		time.Sleep(100 * time.Millisecond) // token is created in a different goroutine
```

Comment:
I see you are too optimistic about our CI machines 😅

Reply:
I'm not a big fan of this approach, can we use `require.Eventually`[1]?

[1] https://pkg.go.dev/github.com/stretchr/testify/require#Eventually